packages/web/src/components/codeMirrorWrapper.tsx

```diff
@@ -90,29 +90,38 @@ export const CodeMirrorWrapper: React.FC<SourceProps> = ({
         lineNumbers,
         lineWrapping: wrapLines,
       });
+      codemirrorRef.current = cm;
       setCodemirror(cm);
-      if (onChange)
-        cm.on('change', () => onChange(cm.getValue()));
-      updateEditor(cm, text, highlight, revealLine, focusOnChange);
       return cm;
     })();
-  }, [modulePromise, codemirror, codemirrorElement, text, language, highlight, revealLine, focusOnChange, lineNumbers, wrapLines, readOnly, onChange]);
+  }, [modulePromise, codemirror, codemirrorElement, language, lineNumbers, wrapLines, readOnly]);
```

Comment:
```suggestion
  }, [modulePromise, codemirror, codemirrorElement, language, lineNumbers, wrapLines, readOnly, codemirrrorRef]);
```

Reply:
codemirrorRef is const, using it in effect does not make sense.